ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/transactional/TransactionalInterceptorBase.java

```diff
@@ -109,9 +109,11 @@ private Transactional getTransactional(InvocationContext ic) {
         // when the CDI implementation is Weld then using the Weld API for accessing the annotated type
         if (ic.getContextData().containsKey(WELD_INTERCEPTOR_BINDINGS_KEY)) {
             Set<Annotation> annotationBindings = (Set<Annotation>) ic.getContextData().get(WELD_INTERCEPTOR_BINDINGS_KEY);
-            for (Annotation annotation : annotationBindings) {
-                if (annotation.annotationType() == Transactional.class) {
-                    return (Transactional) annotation;
+            if (annotationBindings != null) {
```

Comment:
I'm still not sure about the fix. This was the code before the adjustment for handling Weld in other path to fix 
```
[JBTM-3442] using Weld API to get transactional annotation at CDI interceptor if Weld is available
```
-> https://github.com/jbosstm/narayana/blob/5.11.2.Final/ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/transactional/TransactionalInterceptorBase.java#L106

Which works in way: `if CDI -> go the branch otherwhise -> EJB branch`
Your fix does `if Weld -> go the branch -> otherwhise -> do nothing`
I think it should be `if Weld and CDI -> go the branch otherwise check if CDI -> go the branch otherwise -> EJB branch`

I'm still considering the fix should be in way

```
// not-null for CDI && CDI is managed by Weld
if (interceptedBean != null && ic.getContextData().containsKey(WELD_INTERCEPTOR_BINDINGS_KEY)) {
 // when the CDI implementation is Weld then using the Weld API for accessing the annotated type
...
```

Could you please elaborate on your solution for me to undestand if I'm missing something?